dqc_us_rules/dqc_us_0046.py

```diff
@@ -0,0 +1,100 @@
+# (c) Copyright 2015 - 2017, XBRL US Inc. All rights reserved.
+# See https://xbrl.us/dqc-license for license information.
+# See https://xbrl.us/dqc-patent for patent infringement notice.
+import json
+import os
+from arelle import XbrlConst
+from .util import messages
+
+_CODE_NAME = 'DQC.US.0046'
+_RULE_VERSION = '4.0.0'
+_RULE_INDEX_KEY = '1'
+_NO_FACT_KEY = 'no_fact'
+_CONFIG_JSON_FILE = os.path.join(
+            os.path.dirname(__file__),
+            'resources',
+            'DQC_US_0046',
+            'dqc_0046.json'
+        )
+_EMPTY_LIST = []
+
+
+def _find_errors(val):
+    """
+    Entrypoint for the rule.  Load the config, search for instances of
+    reversed calculation relationships.
+
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :return: No direct return
+    :rtype: None
+    """
+    error_list = []
+    config_json_file = _CONFIG_JSON_FILE
+    calc_children = _load_config(config_json_file)
```

Comment:
why not skip the middleman variable and just do `calc_children = _load_config(_CONFIG_JSON_FILE)`?